### PR TITLE
diff parser file lex

### PR DESCRIPTION
**Pull request by windwiny**
racc output file's comment line include version,
use ripper generate lex to compare

### CODE REVIEW

File: test/helper.rb [hunk 1]
```diff
@@ -71,13 +73,17 @@ def assert_parser_unchanged(asset)
...
+        l1 = Ripper.lex(d1).map {|e| e[1,2] unless [:on_comment,:on_nl,:on_ignored_nl,:on_sp].include?(e[1]) }.compact
+        l2 = Ripper.lex(d2).map {|e| e[1,2] unless [:on_comment,:on_nl,:on_ignored_nl,:on_sp].include?(e[1]) }.compact
+        l1 == l2
       end
 
       assert(result, "Output of test/assets/#{asset} differed from " \
         "expectation. Try compiling it and diff with test/regress/#{file}.rb:" \
         "\nruby -I./lib ./bin/racc -o tmp/#{file} test/assets/#{asset}; " \
-        "colordiff tmp/#{file} test/regress/#{file}.rb")
+        "lex_diff tmp/#{file} test/regress/#{file}.rb")
```
Comment (alexdowad) [inline test/helper.rb]: What is `lex_diff`? I've never used it before.

Comment (windwiny) [inline test/helper.rb]: just a name, little like `colordiff`

Comment (alexdowad) [inline test/helper.rb]: `colordiff` is an actual CLI tool. On Ubuntu, you can install it with `sudo apt-get install colordiff`. How about `lex_diff`?

Comment (windwiny) [inline test/helper.rb]: restore it.

**Comment by alexdowad** [comment]
@windwiny I think this is a good idea, but when something breaks the regression tests, we need a way to see exactly where they are broken. To make this easier, I made them print that line of shell (including `colordiff`) which could be cut and pasted into a shell to see exactly where the regression test was failing.

Although you have restored `colordiff` in place of the fictional `lex_diff`, that doesn't really solve the problem. Comparing tokens rather than text bytes makes sense, but if the comparison fails, the test suite should clearly show exactly *where* and *how* it failed.

Can you think of any way to do that?

**Comment by windwiny** [comment]
@alexdowad , I have not idea.
But i think only this gem's (Racc) maintainer need run regression tests. Not the Racc end-users. Maintainer may have enough programming skill to understand assert failed tips.

Or, restore to comparing text, and just skip top 3 comment line on the output file.

**Comment by alexdowad** [comment]
> But i think only this gem's (Racc) maintainer need run regression tests.

Well, anyone who wants to contribute to Racc needs to run them, to make sure they don't break anything.

Racc users don't need to run the test suite at all.

**Comment by alexdowad** [comment]
@windwiny, give me a little bit of time to look at this. I will fix up the test suite so that it compares tokens (not bytes), *and* prints helpful diagnostics if the comparison fails.

File: test/helper.rb [hunk 2]
```diff
@@ -71,7 +73,11 @@ def assert_parser_unchanged(asset)
       file = File.basename(asset, '.y')
 
       result = Dir.chdir(PROJECT_DIR) do
-        File.read("#{REGRESS_DIR}/#{file}.rb") == File.read("#{TAB_DIR}/#{file}")
+        d1 = File.read("#{REGRESS_DIR}/#{file}.rb")
+        d2 = File.read("#{TAB_DIR}/#{file}")
+        l1 = Ripper.lex(d1).map {|e| e[1,2] unless [:on_comment,:on_nl,:on_ignored_nl,:on_sp].include?(e[1]) }.compact
+        l2 = Ripper.lex(d2).map {|e| e[1,2] unless [:on_comment,:on_nl,:on_ignored_nl,:on_sp].include?(e[1]) }.compact
+        l1 == l2
```
Comment (zenspider) [inline test/helper.rb]: Why not use `assert_equal`?

**Comment by zenspider** [comment]
This is beyond stale and the diff is blown out and hard to understand. No response from OP on review. Closing.

**Pull request closed without merging**